internal/app/machined/pkg/controllers/network/cmdline.go

```diff
@@ -255,6 +257,75 @@ func ParseCmdlineNetwork(cmdline *procfs.Cmdline) (CmdlineNetworking, error) {
 			linkSpecSpecs = append(linkSpecSpecs, slaveLinkSpec)
 		}
 	}
+	// dracut vlan=<vlanname>:<phydevice>
+	vlanSettings := cmdline.Get(constants.KernelParamVlan).First()
+
+	if vlanSettings != nil {
+		fields := strings.SplitN(*vlanSettings, ":", 2)
+
+		vlanName := fields[0]
+		phyDevice := fields[1]
+		vlanID, err := func(name string) (uint16, error) {
+			numStart := len(name)
+
+			for i := numStart - 1; i >= 0; i-- {
+				if name[i] >= '0' && name[i] <= '9' {
+					numStart = i
+
+					continue
+				}
+
+				break
+			}
+
+			vlanNumberString := vlanName[numStart:]
+
+			vlanNumber, err := strconv.Atoi(vlanNumberString)
+			if err != nil || vlanNumberString == "" {
+				return 0, errors.New("unable to parse vlan")
+			}
+
+			return uint16(vlanNumber), nil
+		}(vlanName)
+		if err != nil {
```

Comment:
gofumpt does not agree with wsl and I'm unable to satisfy both.

Reply:
I can look into that, and fix that as I re-sign the commit

Reply:
I refactored the code to avoid the situation. I think the code also got easier to read.